### PR TITLE
leetcode #58 and #20 solutions

### PR DESCRIPTION
**Pull request by 0xlmak**
Here is : 
- a solution to leetcode #58 with full test coverage : I don't know if I was allowed to use python built-in functions of string for that one ?
- a solution to leetcode #20 with full test coverage

### CODE REVIEW

File: 20_leet_validparentheses.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from typing import List
+
+
+class Solution:
+    def isValid(self, s: str) -> bool:
+        bracket_pairs: dict[str, str] = {
```
Comment (fspv) [inline 20_leet_validparentheses.py]: Not sure why this works, but it should be:
```python
from typing import Dict
bracket_pairs: Dict[str, str] = {}
```

**Review by fspv — APPROVED** [review]
Looks good to me, just a single nitpick here

**Comment by fspv** [comment]
> I don't know if I was allowed to use python built-in functions of string for that one ?

For easy it is. Though there is a O(1) memory solution for this problem, but let's ignore that for now and return to that once you start solving medium problems.

**Comment by 0xlmak** [comment]
> > I don't know if I was allowed to use python built-in functions of string for that one ?
> 
> For easy it is. Though there is a O(1) memory solution for this problem, but let's ignore that for now and return to that once you start solving medium problems.

Sounds good then

**Pull request merged**